### DISCUSSION TITLE
Windows Terminal v1.18.3181.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#16314] [Announcements]
Welcome to the mid-November servicing release for Windows Terminal!

This release includes the following bug fixes:

### Usability

* Environment variables will no longer spontaneously disappear if one happens to have a `NUL` in it (#16190)
* The "Azure Cloud Shell" connector has been updated to support the new cloud shell protocol (#16247)
* The regression where "Cascadia Mono" (or "... Code") would not be found on upgrade has been un-regressed (#16196)
* The light color schemes now include a selection background that is actually _visible_ (#16243) (thanks @TahaHaksal!)
   * You may need to delete the light color schemes from your settings to see this change.
* With the new rendering engine, some pixel shaders should no longer cause characters to have a black background (#16219)

### Reliability

* Terminal no longer leaks nigh-unto infinite amounts of memory when it's out of focus (#16251)
* When you close a window _on Windows 10_, there's a much lower chance of Terminal blowing up spontaneously (#16267)
* (newly backported from 1.19) A rare crash seen on Windows 10 when creating a new window has been obliterated (#16047)

### Miscellaneous

* The issue where stable-branded binaries contained a `-preview` moniker has been fixed (#16122)
* The system menu no longer grows hundreds of copies of the `Settings...` menu item when you close and reopen a window (#16225)

_Binary files inside the unpackaged distribution archive bear the version number `1.18.231114001`._

---

### Asset Hashes

- Microsoft.WindowsTerminal_1.18.3181.0_8wekyb3d8bbwe.msixbundle_Windows10_PreinstallKit.zip
   - SHA256 `62909C08F5FF9B2504E5B12CC0A5711296C9405F8B2D5E4FE8DD49CC54486D15`
- Microsoft.WindowsTerminal_1.18.3181.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `1C5B383F79E5E5535AB1ADC8DEBF38E5AA4851450E6209FA25BFE81909B8E20D`
- Microsoft.WindowsTerminal_1.18.3181.0_arm64.zip
   - SHA256 `FBB2FB7FF4B16561EFFB03A952CDCC3DAC2FB1DDFC44A3A2BAF276E35319437F`
- Microsoft.WindowsTerminal_1.18.3181.0_x64.zip
   - SHA256 `8BB41F8552A8AB50F9AD0317EE2D7007CF42EE9780C62F9B5A15C1CBD3E4D71A`
- Microsoft.WindowsTerminal_1.18.3181.0_x86.zip
   - SHA256 `5A34110D68559EC83589D1F2D43538D00C4F34B72A538B7ED37D0717CE378F34`

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.18.3181.0'>Windows Terminal v1.18.3181.0</a>.</em>

**Comment by Almighty188** [comment]
When will this version be on Microsoft Store?

**Comment by BurgundyDev** [comment]
Bumping the question by the user above, when's the update coming to MS Store?

**Reply by DHowett** [reply]
It looks like this version failed store certification for publishing to retail. Thanks!

**Reply by Almighty188** [reply]
Thanks, looks like mines actually updated via Microsoft Store on November 21, 2023, a day after I wrote the original comment above while others did not receive it until earlier today.  Does being in Windows 11 Beta Insiders make a difference?

**Reply by DHowett** [reply]
Yep! We publish to the Beta channel a little faster than we publish to GA/Retail. Just to be on the safe side!

**Comment by robsmi-msfte** [comment]
We build custom images for customers, where we don't have Internet access during the build process.  Therefore, we can't automatically update the inbox UWP apps.
Is there a way to update Terminal without connecting to the Store?
(I checked Store for Business and you can install Terminal, but I am not able to find a way to download the .MSIXbundle)

**Comment by jacktose** [comment]
Is there a plan to enable shell integration in the non-preview version?

**Reply by zadjii-msft** [reply]
Yep, we're tracking that in #15057
